### PR TITLE
RadioButtons

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have solved the third task: one page contains three radio buttons that redirect to another pages.

radio buttons

By first accessing:
first page

second accessing

### CODE REVIEW

File: src/com/clouway/buttons/RadioButtonsServlet.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class RadioButtonsServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    PrintWriter writer = resp.getWriter();
+    String paramName = req.getParameter("page");
+
+    if (paramName != null) {
```
Comment (freekman) [inline src/com/clouway/buttons/RadioButtonsServlet.java]: if you have 100 pages what will you do 100 if's?

File: src/com/clouway/helper/HtmlHelper.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      String line;
+      while ((line = reader.readLine()) != null) {
+        builder.append(line);
+      }
+      return builder.toString();
+
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return "";
```
Comment (freekman) [inline src/com/clouway/helper/HtmlHelper.java]: Again add some message

File: test/com/clouway/FirstPageServletTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.ByteArrayOutputStream;
+import java.io.PrintWriter;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class FirstPageServletTest {
+  @Rule
+  public JUnitRuleMockery context = new JUnitRuleMockery();
+  private HttpServletRequest request = context.mock(HttpServletRequest.class);
```
Comment (freekman) [inline test/com/clouway/FirstPageServletTest.java]: .

File: test/com/clouway/FirstPageServletTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  private ByteArrayOutputStream out;
+
+  @Before
+  public void setUp() throws Exception {
+    out = new ByteArrayOutputStream();
+    writer = new PrintWriter(out);
+  }
+
+  @Test
+  public void happyPath() throws Exception {
```
Comment (freekman) [inline test/com/clouway/FirstPageServletTest.java]: what are you testing here

File: test/com/clouway/FirstPageServletTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    context.checking(new Expectations() {{
+      oneOf(response).getWriter();
+      will(returnValue(writer));
+
+      oneOf(request).getSession(true);
+      will(returnValue(session));
+
+      oneOf(session).isNew();
+      will(returnValue(true));
```
Comment (freekman) [inline test/com/clouway/FirstPageServletTest.java]: what will happen if isNew returns false

File: test/com/clouway/RadioButtonsTest.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  private ByteArrayOutputStream out;
+
+  @Before
+  public void setUp() throws Exception {
+    out = new ByteArrayOutputStream();
+    writer = new PrintWriter(out);
+  }
+
+  @Test
+  public void happyPath() throws Exception {
```
Comment (freekman) [inline test/com/clouway/RadioButtonsTest.java]: this is not your happyPath() the ```sendRedirect``` is actually your happyPath()

File: test/com/clouway/FirstPageServletTest.java (hunk 5 above)
Comment (Stanislava88) [inline test/com/clouway/FirstPageServletTest.java]: the message "Welcome" will not be displayed.

File: src/com/clouway/buttons/RadioButtonsServlet.java (hunk 1 above)
Comment (Stanislava88) [inline src/com/clouway/buttons/RadioButtonsServlet.java]: fixed

File: src/com/clouway/helper/HtmlHelper.java (hunk 2 above)
Comment (Stanislava88) [inline src/com/clouway/helper/HtmlHelper.java]: added

File: src/com/clouway/buttons/FirstPageServlet.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    PrintWriter writer = resp.getWriter();
+    HttpSession session = req.getSession(true);
+
+    HtmlHelper helper = new HtmlHelper("web/WEB-INF/first.html");
+    String html = helper.loadResource();
+
+    HtmlTemplate template = new HtmlTemplate(html);
+    template.put("first", "");
```
Comment (freekman) [inline src/com/clouway/buttons/FirstPageServlet.java]: when you see ```template.put("fitst",.....)```` what does this ```first ``` stay for ?

File: src/com/clouway/buttons/FirstPageServlet.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    html = template.evaluate();
+
+    if (session.isNew()) {
+      template.put("first", "WELCOME!!!");
+      html = template.evaluate();
```
Comment (freekman) [inline src/com/clouway/buttons/FirstPageServlet.java]: why is the template evaluated twice

File: src/com/clouway/buttons/FirstPageServlet.java (hunk 7 above)
Comment (Stanislava88) [inline src/com/clouway/buttons/FirstPageServlet.java]: because otherwise in the browser is displayed the template tag.

Comment (freekman) [inline src/com/clouway/buttons/FirstPageServlet.java]: i really dont understand. 
 What is this ```first``` a name , a greeting message or what ?

File: src/com/clouway/buttons/RadioButtonsServlet.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.PrintWriter;
+
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+public class RadioButtonsServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    PrintWriter writer = resp.getWriter();
+    String paramName = req.getParameter("page");
```
Comment (freekman) [inline src/com/clouway/buttons/RadioButtonsServlet.java]: page is a better name

File: test/com/clouway/FirstPageServletTest.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+    firstPage.doGet(request, response);
+
+    String actual = out.toString();
+
+    assertThat(actual, containsString("<div>WELCOME!!!</div>This is first page<br>"));
+  }
+
+  @Test
+  public void secondAccessPage() throws Exception {
```
Comment (freekman) [inline test/com/clouway/FirstPageServletTest.java]: MultipleVisits

File: src/com/clouway/buttons/FirstPageServlet.java (hunk 7 above)
Comment (Stanislava88) [inline src/com/clouway/buttons/FirstPageServlet.java]: yes, this is for greeting message

File: src/com/clouway/buttons/FirstPageServlet.java (hunk 8 above)
Comment (Stanislava88) [inline src/com/clouway/buttons/FirstPageServlet.java]: fixed

File: src/com/clouway/buttons/RadioButtonsServlet.java (hunk 9 above)
Comment (Stanislava88) [inline src/com/clouway/buttons/RadioButtonsServlet.java]: renamed

File: test/com/clouway/FirstPageServletTest.java (hunk 10 above)
Comment (Stanislava88) [inline test/com/clouway/FirstPageServletTest.java]: renamed

File: src/com/clouway/buttons/FirstPageServlet.java (hunk 7 above)
Comment (freekman) [inline src/com/clouway/buttons/FirstPageServlet.java]: ok if i dont know where and what is ```first ``` how can i know this is a greeting message and again for ```second , third ,  fourth , .......................................................```

Comment (Stanislava88) [inline src/com/clouway/buttons/FirstPageServlet.java]: renamed to greeting

Comment (freekman) [inline src/com/clouway/buttons/FirstPageServlet.java]: huraaaaa

**Comment by freekman** [comment]
LGTM

**Comment by stefan-dimitrov** [comment]
LGTM.

**Pull request merged**